pkg/controller/machine.go

```diff
@@ -614,6 +651,23 @@ func (c *controller) machineDelete(machine *v1alpha1.Machine, driver driver.Driv
 			}
 
 			err = driver.Delete(machineID)
+		} else {
+			klog.V(2).Infof("Missing ProviderID when deleting the machine object. %s might have been deleted too soon.", machine.Name)
+			// As MachineID is missing, we should check once if actual VM was created but MachineID was not updated on machine-object.
+			// We list VMs and check if any one them map with the given machine-object.
+			VMList, _ := driver.GetVMs("")
```

Comment:
Same as above. Better not to ignore any error here.

Reply:
Yes, it makes sense. Thanks for catching this, I shall handle the error here.

Reply:
done